### ISSUE TITLE
[5.2]Wrong locale on installation

### ISSUE THREAD

**Issue by dautrich** [joomla/joomla-cms#44667]
### Steps to reproduce the issue

1. Make sure your Windows and your browser are set to locale “Deutschland” (or de-DE, where applicable).
2. Try to install a J5 version, i. e. 5.2.2

### Expected result

The installation language is displayed as “German (Germany) | Deutsch (Deutschland)”.

### Actual result

The installation language is displayed as “German (Austria) | Deutsch (Österreich)”.

![Image](https://github.com/user-attachments/assets/fb1af421-8478-4e3b-a09c-086ee98def1a)

### System information (as much as possible)

Windows 11 (Home or Pro)
Firefox 133.0.3 (or earlier) or Microsoft Edge 131.0.2903.112 (or earlier)

### Additional comments
**No issue** with Google Chrome (tested on 131.0.6778.205)

I have seen set since the first J5 releases and already had a discussion with @tecpromotion about that topic in February 2024.

**Comment by tecpromotion** [comment]
Sorry but I can't reproduce this in my local test environment with 5.2.3 RC 1.
It works as expected here.

Tested with Chrome, Firefox and Safari.

**Comment by fgsw** [comment]
Can't  confirm too (Mac, Firefox).

**Comment by brianteeman** [comment]
@tecpromotion I assume as you tested with safari that you are like @fgsw testing on a Mac and not the reported issue on Windows

**Comment by tecpromotion** [comment]
That's right. But I have already written to Rolf. We will try to reproduce this in the German-speaking community. Maybe we'll find out something.

**Comment by richard67** [comment]
@dautrich @tecpromotion @fgsw Could you inspect the request header for the "GET" request in the network traffic analysis of your browser's developer tools?

There should be an `Accept-Language` header.

See e.g. screenshot from my browser (German Firefox latest version on Windows):

![Image](https://github.com/user-attachments/assets/1b239be9-ced1-40ee-a5ee-5d8b4890d779)

In my case the value is `de-DE,de;q=0.8,en-US;q=0.5,en;q=0.3`, and I cannot reproduce the issue on a current 5.2-dev branch.

You can test that with any GET request to any (valid) URL.

Please check and report back your values for that `Accept-Language` header.

**Comment by richard67** [comment]
Same result (issue not reproduced) with Chrome, however with a different value for the header, compared to Firefox:
![Image](https://github.com/user-attachments/assets/9319f8a9-f01a-4f94-867a-58669a954f00)

And same (issue not reproduced with different header compared to Firefox or Chrome) with Edge:
![Image](https://github.com/user-attachments/assets/64217e97-8378-4ced-83c7-8dd0fc273784)

**Comment by fgsw** [comment]
@richard67 Firefox is set to English, for this test preferred Language on "German [de]", give:

![Image](https://github.com/user-attachments/assets/5d24311f-da13-4aa0-988e-d8e8817edf49)

**Comment by richard67** [comment]
@fgsw Hmm, so in your case it finds the right language even if your header has no "de-DE", only "de". The ordering of the list field in the form is so that "de-AT" comes before "de-DE", so it seems in your case it does not just pick the first "de-something" which it can find. Strange. So we have to wait what @dautrich reports for the case in which he can reproduce the issue.

**Comment by dautrich** [comment]
Accept-Language | de,en-US;q=0.7,en;q=0.3

Screenshot from my Firefox:    
![Image](https://github.com/user-attachments/assets/0837dbf1-ba5a-4549-91ce-7e87d3fc189a)

**Comment by dautrich** [comment]
Microsoft Edge:     accept-language: de,de-DE;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6

![Image](https://github.com/user-attachments/assets/89f78eb0-65e6-4e51-86bb-a776df6a2b32)

**Comment by richard67** [comment]
@dautrich So in your case as well as in @fgsw 's you don't have a "de-DE", only "de", but in opposite to him you have "en-US".

What happens if you either modify the header in the developer tools or if you change the settings as shown in following screenshots?

![Image](https://github.com/user-attachments/assets/cf41007b-fe73-4ce7-a4d3-72a80110f940)

![Image](https://github.com/user-attachments/assets/88f988c4-14c2-4e43-814e-b457a092595d)

I think in your case the one at the top is missing. But you can add it and move it to the top with that dialogue.

**Comment by dautrich** [comment]
That does the trick: I get    Accept-Language | de-DE,de;q=0.8,en-US;q=0.5,en;q=0.3
resulting in that:

![Image](https://github.com/user-attachments/assets/d92961c1-4572-401b-a8b3-633453ca64b4)

**Comment by richard67** [comment]
So that is at least a feasible workaround. If we have an issue or not I don’t really know. But it seems weird to me that in your case with the issue de-AT was used and in @fgsw ‘s case de-DE.

It would need to check the code to see how it works.

**Comment by richard67** [comment]
P.S. I don’t really remember if the de-DE language was in my Firefox settings by default or if I once had added it myself.

**Comment by dautrich** [comment]
The same mechanism works in Microsoft Edge.
When I move "Deutsch (Deutschland)" to the top of the list, I get my preferred language setting:

![Image](https://github.com/user-attachments/assets/99d3fc15-c5c7-4626-9063-2d606734dddb)

**Comment by dautrich** [comment]
@richard67 Would it be possible to default to "Deutsch (Deutschland)", if the first language in the Accept-Language list is "de"?

**Comment by richard67** [comment]
That’s what I would expect.

But the question is what shall happen if only de is given and not de-DE:

Shall it take the first de-something which it can find in alphabetical order? Then it would be de-AT.

Or shall it take the primary country associated with a particular language? That would be de-DE in case of German, and for English orbother languages it would be debatable, and that would require a list of what is the primary country for each language.

**Comment by richard67** [comment]
> [@richard67](https://github.com/richard67) Would it be possible to default to "Deutsch (Deutschland)", if the first language in the Accept-Language list is "de"?

That would require some kind of a hard-coded list to define the primary country (or language with country) for each language in thevonebor the other way.

And e.g. in case of English, where in opposite to German the country with the most speakers is not the same as the country where the language comes from, it would be debatable if GB or US should be the country.

**Comment by dautrich** [comment]
IMHO, there are two aspects:

1. Is it technically possible (with not too much effort) to set a "default" language for those ambiguous cases (as far as I can see: Dutch, English, French, German, Portuguese)?
2. If technically possible: Who is to decide for the default setting. The Languages Team?

**Comment by richard67** [comment]
Sure it is technically possible. So 2. is the interesting question: Who decides?

**Comment by tecpromotion** [comment]
Chrome on Mac
![Image](https://github.com/user-attachments/assets/6dc7675f-ae36-4e54-85ab-255ca7a6202b)

Safari on Mac
![Image](https://github.com/user-attachments/assets/f475b13b-ffae-41a8-b412-8dc621d6f271)

Firefox on Mac
![Image](https://github.com/user-attachments/assets/a10f79af-2d4c-44e3-be11-a66f77f0739f)

**Comment by richard67** [comment]
In case of English, most software I know uses US as default because that has meanwhile become the new standard in English lessons in the world and has more speakers, as it seems. In my school days Oxford English was the standard.

But as we use en-GB as default CMS language, I would suggest to default to en-GB to be consistent with that and not to en-US, if we decide to implement a solution.

**Comment by richard67** [comment]
> as far as I can see: Dutch, English, French, German, Portuguese

No, that does not cover all cases. A solution should cover all possible codes available in standards, so I does not need to update it whenever a new language is added to the CMS.

For example Farsi is spoken in Iran and in Pakistan, Urdu is spoken in Pakistan and I think also in parts of Afghanistan, Hindu or Tamil are spoken not only in India, and so on.

**Comment by brianteeman** [comment]
> The same mechanism works in Microsoft Edge. When I move "Deutsch (Deutschland)" to the top of the list, I get my preferred language setting:
> 
> ![Image](https://github.com/user-attachments/assets/78e9e44c-cf97-464d-a055-fb8b96bdce64)

So the joomla code is working exactly as intended

**Comment by fgsw** [comment]
> So the joomla code is working exactly as intended

Correct. 

@richard67 If for Example "German [de]" is set as preferred Language, i suggest a "easy as possible" solution which seems the alphabetic odering.

**Comment by chmst** [comment]
Other question: This affects only the instalation process, nothing else. 
I cold not find any words in the de-AT version which I could not understand during the installation. This probably is alos for other languanges. 
So why should we should make efforts here?

**Comment by richard67** [comment]
Well, I'm ok with leaving it as it is. My above comments were just for the case that there is common sense to do something.

@dautrich Ok to close the issue as expected behaviour?

**Comment by dautrich** [comment]
In principle, I agree to @chmst's post. However, it's a bit strange for a Joomla user to be offered a (for him) wrong locale at his very first contact with his new Joomla site. 
Could we just add a note to the installation screen? Something like: “In case you're offered a wrong region setting for your language, that is caused by your browser's settings.”, together with a link to a page with some more explanations (i.e. using screenshots like above) on the Joomla site.

And @richard67: I'm okay with defining that as expected behavior. We now know the reason for this effect and how to cope with it.

**Comment by fgsw** [comment]
> Could we just add a note to the installation screen?

Please don't clutter the installation pages. 

If you wan't add a note on the Docs.

**Comment by fgsw** [comment]
> Could we just add a note to the installation screen?

Please don't clutter the installation pages. 

If you wan't add a note on [Docs: Installing Joomla](https://docs.joomla.org/J4.x:Installing_Joomla#Start_Install) in the second Paragraph:

>Joomla will try to identify the _Select Language_ field automatically from your browser's language. You can change this if needed.

**Comment by dautrich** [comment]
> Please don't clutter the installation pages.

That obviously is a good statement from the technical point of view. But my point is about psychology: Think about a newbie at his first Joomla installation. The very first thing he/she sees may be a "wrong" locale identification. This won't increase the user's confidence in the new software. That's why I would like to have a remark on the installation page. You and I will simply ignore the message.

**Comment by chmst** [comment]
Why not replace the de-AT by either simple "de" or even better "German"?

**Comment by fgsw** [comment]
> Why not replace the de-AT by either simple "de" or even better "German"?

Good Solution cause it solves another issue too:
 If a user wan't to have a site in "Portuguese" but not for a spcific Country he have to choose a country like "Portugal" or "Brazil" with a country flag (if he hide the country-flags Joomla show "pt-PT" or "pt-BR").

**Comment by dautrich** [comment]
Correct settings in the user's browser (i.e. Firefox, Edge) avoids this issue. PR #44735 adds a better solution for J6. 
Closing this.
